lib/src/widgets/chat_list.dart

```diff
@@ -0,0 +1,131 @@
+    this.order = GroupedListOrder.ASC,
+    this.sort = true,
+    this.inComingChatBubbleBorderRadius,
+    this.outgoingChatBubbleBorderRadius,
+  }) : super(key: key);
+
+  final ScrollController scrollController;
+  final bool horizontalDragToShowTime;
+  final List<Message> messageList;
+  final GroupedListOrder order;
+  final bool sort;
+  final Widget Function(String separator)? groupSeparatorBuilder;
+  final String currentUser;
+  final Color? backgroundColor;
+  final String? backgroundImageFromAsset;
+  final String? backgroundImageFromNetwork;
+  final Color? outgoingChatBubbleColor;
+  final Color? inComingChatBubbleColor;
+  final TextStyle? chatBubbleTextStyle;
+  final TextStyle? linkStyle;
+  final void Function()? onShareButtonPressed;
+  final void Function()? onImageViewTap;
+  final Icon? shareIcon;
+  final Color? shareIconBackgroundColor;
+  final double? heightOfImageMessage;
+  final double? widthOfImageMessage;
+  final BorderRadiusGeometry? inComingChatBubbleBorderRadius;
+  final BorderRadiusGeometry? outgoingChatBubbleBorderRadius;
+
+  @override
+  _ChatListState createState() => _ChatListState();
+}
+
+class _ChatListState extends State<ChatList> {
+  @override
+  Widget build(BuildContext context) {
+    final isBackgroundImage = widget.backgroundImageFromAsset != null ||
+        widget.backgroundImageFromNetwork != null;
+    return Container(
+      height: MediaQuery.of(context).size.height,
+      decoration: BoxDecoration(
+        color: widget.backgroundColor ?? Colors.transparent,
+        image: isBackgroundImage
+            ? DecorationImage(
+                fit: BoxFit.fill,
+                image: widget.backgroundImageFromAsset != null
+                    ? AssetImage(widget.backgroundImageFromAsset!)
+                        as ImageProvider
+                    : NetworkImage(widget.backgroundImageFromNetwork!),
+              )
+            : null,
+      ),
+      child: Stack(
+        children: [
+          SingleChildScrollView(
+            controller: widget.scrollController,
+            child: Column(
+              children: [
+                GroupedListView<Message, String>(
+                  shrinkWrap: true,
+                  elements: widget.messageList,
+                  groupBy: (element) => element.createdAt.getDateFromDateTime,
+                  itemComparator: (message1, message2) =>
+                      message1.message.compareTo(message2.message),
+                  physics: const NeverScrollableScrollPhysics(),
+                  order: widget.order,
+                  sort: widget.sort,
+                  groupSeparatorBuilder: (separator) =>
+                      widget.groupSeparatorBuilder != null
+                          ? widget.groupSeparatorBuilder!(separator)
+                          : ChatGroupHeader(day: DateTime.parse(separator)),
+                  indexedItemBuilder: (context, message, index) => ChatWidget(
```

Comment:
change the name of the chat widget to the appropriate name if it's a chat bubble then name it accordingly.